### PR TITLE
Update GHC

### LINKED ISSUE #13915: Can't implement two interfaces with the same (unqualified) name.
# Bug description

If you try to implement two interfaces that have the same (unqualified) name, in the same template, the desugarer spits out duplicate bindings, which causes errors. Additionally if the two interfaces have a method with the same name, this causes more duplicate bindings.

# To reproduce

Module A:

```
module A where
interface I where
   myMethod : Party
```

Module B:

```
module B where
interface I where
   myMethod : Party
```

Module C:

```
module C where
import qualified A
import qualified B
template T with
   p : Party 
 where
   signatory p
   implements A.I where
      myMethod = p
   implements B.I where
      myMethod = p
```

If you try to build C, you get an error like:

```
<no location info>: error:
    Duplicate type signatures for '_implements_T_I'
    at <no location info>
       <no location info>typecheck
<no location info>: error:
    Multiple declarations of '_implements_T_I'
    Declared at: <no location info>
                 <no location info>typecheck
```

Additionally, because of the method names, you get:

```
<no location info>: error:
    Duplicate type signatures for '_method_T_I_myMethod'
    at <no location info>
       <no location info>typecheck

<no location info>: error:
    Multiple declarations of '_method_T_I_myMethod'
    Declared at: <no location info>
                 <no location info>typecheck
```

# Expected behavior

No errors.

# Additional context

This happens because the desugarer isn't taking into account that `A.I` and `B.I` appear qualified in the implements block, when generating the `_implements_` and `_method_` bindings.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
with https://github.com/digital-asset/ghc/pull/120

fixes #13915

part of #13653

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: bazel_tools/ghc-lib/version.bzl [hunk 1]
```diff
@@ -9,7 +9,7 @@ GHC_LIB_PATCHES = [
 ]
 
 GHC_REPO_URL = "https://github.com/digital-asset/ghc"
-GHC_REV = "b166432c45a54eaeefb9bcb61523b293da812825"
+GHC_REV = "065554992722bc41fb985be8d109ac66a981f200"
```
Comment (remyhaemmerle-da) [inline bazel_tools/ghc-lib/version.bzl:12]:
```suggestion
GHC_REV = "dd929af4af743e01ab975435420c1a568907a086"
```

**Comment by remyhaemmerle-da** [comment]
@sofiafaro-da @moisesackerman-da could you have a look at the test update before I merge digital-asset/#120

File: compiler/damlc/tests/daml-test-files/InterfaceChoiceCollision2.daml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- @SINCE-LF-FEATURE DAML_INTERFACE
+module InterfaceChoiceCollision2 where
```
Comment (remyhaemmerle-da) [inline compiler/damlc/tests/daml-test-files/InterfaceChoiceCollision2.daml:5]: This test confirm there is not collision check for requiring interfaces.

**Review by akrmn — APPROVED** [review]
Looks good to me, though I think some other `*.EXPECTED.desugared-daml` tests will break. You can update them with 
```bash
bazel run //compiler/damlc/tests:daml-desugar -- --accept
```

**Comment by remyhaemmerle-da** [comment]
>  I think some other *.EXPECTED.desugared-daml tests will break

Shouldn't the changes impact only imported Interfaces ?

File: compiler/damlc/tests/daml-test-files/InterfaceChoiceCollision2.daml (hunk 2 above)
Comment (sofiafaro-da) [reply compiler/damlc/tests/daml-test-files/InterfaceChoiceCollision2.daml:5]: > This test confirms there is not collision check for requiring interfaces.

Can you add this as a comment in the test?

File: compiler/damlc/tests/daml-test-files/InterfaceChoiceCollision.daml [hunk 3]
```diff
@@ -2,41 +2,17 @@
 -- SPDX-License-Identifier: Apache-2.0
 
 -- @SINCE-LF-FEATURE DAML_INTERFACE
--- @ERROR name collision between choice InterfaceChoiceCollision:T.MyArchive (via interface InterfaceChoiceCollision:InterfaceA) and choice InterfaceChoiceCollision:T.MyArchive (via interface InterfaceChoiceCollision:InterfaceB)
+-- @ERROR name collision between choice InterfaceChoiceCollision:T.MyArchive (via interface InterfaceChoiceCollision1:Interface) and choice InterfaceChoiceCollision:T.MyArchive (via interface InterfaceChoiceCollision2:Interface)
 module InterfaceChoiceCollision where
```
Comment (sofiafaro-da) [inline compiler/damlc/tests/daml-test-files/InterfaceChoiceCollision.daml:6]: Can you link the issue #13915 in a comment, so we don't forget this is also a regression test.

**Review by sofiafaro-da — APPROVED** [review]
LGTM, thank you!

**Comment by akrmn** [comment]
> Shouldn't the changes impact only imported Interfaces ?

Ah sorry, you're right, I forgot that this doesn't add the original module but rather the qualifier used in the importing module. Thanks for bearing with me

File: compiler/damlc/tests/daml-test-files/InterfaceChoiceCollision2.daml (hunk 2 above)
Comment (remyhaemmerle-da) [reply compiler/damlc/tests/daml-test-files/InterfaceChoiceCollision2.daml:5]: > Can you add this as a comment in the test?

I will literally drop the collision check in the next PR, I assumed it was not necessary.

**Pull request merged**